### PR TITLE
Code clean up 4

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?
- Reworked plugin & devfile registry synchronization 

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/master/README.md#updating-custom-resource-definition-file)
- [ ] [Nightly OLM bundle is up to date](https://github.com/eclipse-che/che-operator/blob/master/README.md#update-nightly-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: pkg/controller/che/che_controller.go [hunk 1]
```diff
@@ -831,23 +797,44 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 		}
 	}
 
-	provisioned, err = devfile_registry.SyncDevfileRegistryToCluster(deployContext)
-	if !tests {
-		if !provisioned {
+	if !instance.Spec.Server.ExternalPluginRegistry {
+		pluginregistry := registry.NewPluginRegistry(deployContext)
```
Comment (AndrienkoAleksandr) [inline pkg/controller/che/che_controller.go:801]: Minor: pluginregistry -> pluginRegistry

File: pkg/controller/che/che_controller.go [hunk 2]
```diff
@@ -831,23 +797,44 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
...
 			}
-			return reconcile.Result{RequeueAfter: time.Second * 1}, err
 		}
 	}
 
-	provisioned, err = plugin_registry.SyncPluginRegistryToCluster(deployContext)
-	if !tests {
-		if !provisioned {
+	if !instance.Spec.Server.ExternalDevfileRegistry {
+		devfileregistry := registry.NewDevfileRegistry(deployContext)
```
Comment (AndrienkoAleksandr) [inline pkg/controller/che/che_controller.go:819]: Minor: devfileregistry -> devfileRegistry

File: pkg/deploy/postgres/postgres_deployment.go [hunk 3]
```diff
@@ -0,0 +1,206 @@
+//
+// Copyright (c) 2012-2019 Red Hat, Inc.
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/postgres/postgres_deployment.go:2]: Minor: 2019-2021

File: pkg/deploy/registry/devfileregistry.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package registry
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/registry/devfileregistry.go:12]: maybe we should rename file devfileregistry.go  -> devfile_registry.go , f.e https://github.com/golang/go/blob/master/src/crypto/subtle/constant_time.go

File: pkg/deploy/registry/pluginregistry.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package registry
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/registry/pluginregistry.go:12]: Maybe we should rename pluginregistry.go -> plugin_registry.go

File: pkg/deploy/registry/pluginregistry_test.go [hunk 6]
```diff
@@ -0,0 +1,85 @@
+//
+// Copyright (c) 2012-2019 Red Hat, Inc.
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/registry/pluginregistry_test.go:2]: Minor 2019 -> 2021

File: pkg/deploy/server/che_configmap.go [hunk 7]
```diff
@@ -175,33 +174,27 @@ func GetCheConfigMapData(deployContext *deploy.DeployContext) (cheEnv map[string
 	var keycloakInternalURL, pluginRegistryInternalURL, devfileRegistryInternalURL, cheInternalAPI, webSocketEndpoint, webSocketEndpointMinor string
 
 	if deployContext.CheCluster.Spec.Server.UseInternalClusterSVCNames && !deployContext.CheCluster.Spec.Auth.ExternalIdentityProvider {
-		keycloakInternalURL = deployContext.InternalService.KeycloakHost
+		keycloakInternalURL = fmt.Sprintf("%s://%s.%s.svc:8080/auth", "http", deploy.IdentityProviderName, deployContext.CheCluster.Namespace)
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/server/che_configmap.go:177]: Actually the main idea why I didn't evaluated internal url info in the che_configmap.go it was - apply url info after deploying component. But up to you.

**Comment by AndrienkoAleksandr** [comment]
Maybe we should create postgre, plugin and devfile registries objects and another once https://github.com/eclipse-che/che-operator/blob/master/pkg/controller/che/che_controller.go#L82 . But that's only proposal, up to you.

**Comment by tolusha** [comment]
> Maybe we should create postgre, plugin and devfile registries objects and another once

Yes, That's my ultimate goal.
But only when I refactor server and identityprovider deployment.

File: pkg/deploy/registry/init_test.go [hunk 8]
```diff
@@ -9,7 +9,7 @@
 // Contributors:
 //   Red Hat, Inc. - initial API and implementation
 //
-package plugin_registry
+package registry
```
Comment (mmorhun) [inline pkg/deploy/registry/init_test.go:12]: The same as above: we should distinguish between plugin and devfile registries, imo.

File: pkg/deploy/registry/devfileregistry.go [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+	if !done {
+		return false, err
+	}
+
+	done, err = p.SyncConfigMap()
+	if !done {
+		return false, err
+	}
+
+	done, err = p.SyncDeployment()
```
Comment (mmorhun) [inline pkg/deploy/registry/devfileregistry.go:50]: When do we need to use separate methods instead of `SyncAll`?

Comment (tolusha) [reply pkg/deploy/registry/devfileregistry.go:50]: - it allows to test them separately if needed
- it looks clearer

File: pkg/deploy/registry/init_test.go (hunk 8 above)
Comment (tolusha) [reply pkg/deploy/registry/init_test.go:12]: Moved to different packages.

File: pkg/deploy/registry/devfileregistry.go (hunk 9 above)
Comment (mmorhun) [reply pkg/deploy/registry/devfileregistry.go]: Also it makes think that it is ok to use, say, `SyncConfigMap` on its own

**Review by mmorhun — APPROVED** [review]

**Pull request merged**